posts/2024/farmOS-3.0.0/index.md

```diff
@@ -0,0 +1,148 @@
+farmOS), is that farmOS v3 updated from Drupal 9 to
+[Drupal 10](https://www.drupal.org/project/farm/issues/3330490). This also means
+the underlying PHP [Symfony](https://symfony.com) framework was updated from v4
+to v5. In the grand scheme of things, these are routine dependency updates, and
+don't mean a lot to most farmOS users. But, if you are a farmOS module
+developer, it may mean breaking changes to your code, so semantic versioning
+rules dictate that we need to increment the farmOS major version to communicate
+that.
+
+The Simple OAuth module has been updated to version 6. This includes a few
+breaking changes which may affect API integrations. farmOS includes code to
+handle the transition of its own OAuth clients and scopes, but if you have made
+any additional clients that used special roles they will also need to be
+updated. The biggest changes are that the "Implicit" grant type has been
+removed, and the "Password Credentials" grant type has been moved to an optional
+"Simple OAuth Password Grant" module, which must be enabled in order to use that
+grant type. The default farmOS client that is included with farmOS has also been
+moved to a separate module that is not enabled by default. After the update to
+farmOS 3.x, all access tokens will be invalidated, but refresh tokens will still
+work to get a new access token.
+
+Another notable change is that the farmOS v1->v2 migration code has been removed
+in v3. This means that if you are still running farmOS v1 you will not be able
+to update directly from v1 to v3. You will need to
+["migrate" from v1 to v2](https://farmOS.org/hosting/migration/), and then
+"update" from v2 to v3 following the normal
+[farmOS update guide](https://farmos.org/hosting/update/).
+
+The rest of the changes are relatively small, but if you are a developer of any
+farmOS integrations or modules be sure to read the full release notes to see if
+any changes might affect your code.
+
+[3.0.0-beta1 release notes](https://github.com/farmOS/farmOS/blob/3.x/CHANGELOG.md#300-beta1-2023-11-01)
+
+## A look back on v2
+
+We made a lot of progress in the farmOS v2 release cycle (apart from the big
+[changes between v1 and v2](../2023/farmOS-2.0.0)). It's important to look back
+and acknowledge how far we've come! Here is a quick summary of the notable
+farmer-facing features that were added in farmOS v2:
+
+- Planting, Birth, Movement, and Group Membership quick forms.
+- CSV importers for assets, logs, and taxonomy terms.
+- Asset ownership assignment.
+- Bulk actions for log categorization and assigning asset parents.
+- Lab test quantity type with "Test method" field, and "Tissue" lab test type.
+- Localization module for enabling translations.
+- Lots of small user interface/experience (UI/UX) features and improvements.
+
+And that doesn't include all the developer-facing features and improvements.
+For the full list, refer to the
+[farmOS Chanagelog](https://github.com/farmOS/farmOS/blob/3.x/CHANGELOG.md).
```

Comment:
```suggestion
[farmOS Changelog](https://github.com/farmOS/farmOS/blob/3.x/CHANGELOG.md).
```

Reply:
Oops good eye @wotnak. I force pushed the commit with this fixed.